browser/ui/webui/brave_rewards_internals_ui.cc

```diff
@@ -0,0 +1,176 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_rewards_internals_ui.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/json/json_writer.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "brave/components/brave_rewards/browser/rewards_service_factory.h"
+#include "brave/components/brave_rewards/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/render_view_host.h"
+#include "content/public/browser/web_ui.h"
+#include "content/public/browser/web_ui_message_handler.h"
+
+#if !defined(OS_ANDROID)
+#include "brave/components/brave_rewards/resources/grit/brave_rewards_generated_map.h"
+#include "brave/components/brave_rewards/resources/grit/brave_rewards_resources.h"
+#else
+#include "components/brave_rewards/settings/resources/grit/brave_rewards_settings_generated_map.h"
+#include "components/grit/components_resources.h"
+#include "components/grit/components_scaled_resources.h"
+#endif
+
+namespace {
+
+std::string BooleanToString(bool value) {
+  return value ? "true" : "false";
+}
+
+class RewardsInternalsDOMHandler : public content::WebUIMessageHandler {
+ public:
+  RewardsInternalsDOMHandler(
+      base::WeakPtr<BraveRewardsInternalsUI> rewards_internals_ui);
+  ~RewardsInternalsDOMHandler() override;
+
+  // WebUIMessageHandler implementation.
+  void RegisterMessages() override;
+
+ private:
+  void HandleRefresh(const base::ListValue* args);
+
+  base::WeakPtr<BraveRewardsInternalsUI> rewards_internals_ui_;
+
+  DISALLOW_COPY_AND_ASSIGN(RewardsInternalsDOMHandler);
+};
+
+RewardsInternalsDOMHandler::RewardsInternalsDOMHandler(
+    base::WeakPtr<BraveRewardsInternalsUI> rewards_internals_ui)
+    : rewards_internals_ui_(rewards_internals_ui) {}
+
+RewardsInternalsDOMHandler::~RewardsInternalsDOMHandler() {}
+
+void RewardsInternalsDOMHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "brave_rewards_internals.refresh",
+      base::BindRepeating(&RewardsInternalsDOMHandler::HandleRefresh,
+                          base::Unretained(this)));
+}
+
+void RewardsInternalsDOMHandler::HandleRefresh(const base::ListValue* args) {
+  if (!rewards_internals_ui_)
+    return;
+  rewards_internals_ui_->UpdateWebUIProperties();
```

Comment:
I'm confused, this doesn't call `GetRewardsInternalsInfo` so nothing is ever updated. You need a RegisterMessageCallback for `brave_rewards_internals.getRewardsInternals` and the handler should call `GetRewardsInternalsInfo`. The `OnGetRewardsInternalsInfo` callback should call `CallJavascriptFunctionUnsafe` to send the data back to the page